api/src/main/java/org/apache/iceberg/io/CloseableGroup.java

```diff
@@ -23,20 +23,47 @@
 import java.io.IOException;
 import java.util.Deque;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class CloseableGroup implements Closeable {
-  private final Deque<Closeable> closeables = Lists.newLinkedList();
+  private static final Logger LOG = LoggerFactory.getLogger(CloseableGroup.class);
 
-  protected void addCloseable(Closeable closeable) {
+  private final Deque<AutoCloseable> closeables = Lists.newLinkedList();
+  private boolean suppressCloseFailure = false;
+
+  protected void addCloseable(AutoCloseable closeable) {
     closeables.add(closeable);
   }
 
+  protected void setSuppressCloseFailure(boolean shouldSuppress) {
+    this.suppressCloseFailure = shouldSuppress;
+  }
+
   @Override
   public void close() throws IOException {
     while (!closeables.isEmpty()) {
-      Closeable toClose = closeables.removeFirst();
+      AutoCloseable toClose = closeables.removeFirst();
       if (toClose != null) {
-        toClose.close();
+        if (suppressCloseFailure) {
```

Comment:
Personally I think the caller should be handling errors. Why suppress this at the source?

Reply:
Thanks for the quick review! 

This is more of a helper utility function to me and on opt-in basis. In the case we were looking at (aws catalogs), if one of the resource fails to be closed, we don't really plan to handle it further, and we want to make sure an earlier close failure shouldn't block other resources from getting closed. So we either do something like 
```
try {
  resource1.close();
} catch (Exception e) {
  log.error("Error closing resource1, suppressing");
}

try {
  resource2.close();
} catch (Exception e) {
  log.error("Error closing resource2, suppressing");
}
```
or reuse this class to make the code simpler (the current solution).

Reply:
That makes sense. I think it should be a static utility rather than part of an abstract class.